src/runtime_src/core/common/device.h

```diff
@@ -243,6 +243,8 @@ class device
 
   virtual void read(uint64_t offset, void* buf, uint64_t len) const = 0;
   virtual void write(uint64_t offset, const void* buf, uint64_t len) const = 0;
+  virtual int  open(const std::string& subdev, int flag) const = 0;
+  virtual void close(int dev_handle) const = 0;
```

Comment:
Please comment on these functions, plus the two above.  This is an interface, any new functions we add to the interface must be documented, .e.g. I have no idea what they are doing, or why they are needed?

Reply:
Will do. We need open to get an fd so that we can do non-pcie bar read/write on it. This is being used in xospiversal and now in xspi as well